### PR TITLE
[Kotlin][server] Add kotlin-vertx-server

### PR DESCRIPTION
**Pull request by Wooyme**
### Description of the PR

add vertx server for kotlin

### CODE REVIEW

File: modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/KotlinVertxServerCodegen.java [hunk 1]
```diff
@@ -0,0 +1,214 @@
+package org.openapitools.codegen.languages;
```
Comment (wing328) [inline modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/KotlinVertxServerCodegen.java]: @Wooyme please add the copyright note below

https://github.com/OpenAPITools/openapi-generator/blob/master/modules/openapi-generator/src/test/java/org/openapitools/codegen/fsharp/FSharpServerCodegenTest.java#L1-L15

File: modules/openapi-generator/src/main/resources/META-INF/services/org.openapitools.codegen.CodegenConfig [hunk 2]
```diff
@@ -107,3 +107,5 @@ org.openapitools.codegen.languages.TypeScriptJqueryClientCodegen
 org.openapitools.codegen.languages.TypeScriptNodeClientCodegen
 org.openapitools.codegen.languages.TypeScriptRxjsClientCodegen
 org.openapitools.codegen.languages.FsharpGiraffeServerCodegen
+
+org.openapitools.codegen.languages.KotlinVertxServerCodegen
```
Comment (wing328) [inline modules/openapi-generator/src/main/resources/META-INF/services/org.openapitools.codegen.CodegenConfig]: @Wooyme Can you put this according to alphabetical order?

Please move `FsharpGiraffeServerCodegen` (line 109) to the proper location in this file based on alphabetical order if you don't mind.

File: modules/openapi-generator/src/main/resources/kotlin-vertx-server/README.mustache [hunk 3]
```diff
@@ -0,0 +1,3 @@
+{{^hideGenerationTimestamp}}
+    Project generated on : {{generatedDate}}
+{{/hideGenerationTimestamp}}
```
Comment (wing328) [inline modules/openapi-generator/src/main/resources/kotlin-vertx-server/README.mustache]: Can we create the README.mustache based on https://github.com/OpenAPITools/openapi-generator/blob/master/modules/openapi-generator/src/main/resources/kotlin-server/README.mustache to start with?

**Comment by wing328** [comment]
Thanks for the PR. 

Can you please ping me via https://gitter.im (ID: wing328) as I've some suggestions?

**Comment by wing328** [comment]
cc Kotlin tech committee: @jimschubert (2017/09) @dr4ke616 (2018/08) @karismann (2019/03) @Zomzog (2019/04)

**Comment by jimschubert** [comment]
Generated code doesn't seem to build.

```
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  5.891 s
[INFO] Finished at: 2019-08-10T21:32:41-04:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal on project openapi-kotlin-vertx-server: Could not resolve dependencies for project org.openapitools:openapi-kotlin-vertx-server:jar:1.0.0-SNAPSHOT: Could not find artifact com.github.wooyme:vertx-openapi-router:jar:1.0.2 in central (https://repo.maven.apache.org/maven2) -> [Help 1]
[ERROR]
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR]
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/DependencyResolutionException
```

**Comment by Wooyme** [comment]
Sorry,I forgot upload 1.0.2 to maven central. It's ok now.

**Comment by wing328** [comment]
I got errors when running `mvn clean package` in the auto-generated code:
```
[WARNING] Using experimental Kotlin incremental compilation
[ERROR] /Users/williamcheng/Code/openapi-generator/samples/server/petstore/kotlin/vertx/src/main/kotlin/org/openapitools/server/api/verticle/PetApiVertxProxyHandler.kt: (103, 24) Unresolved reference: status
[ERROR] /Users/williamcheng/Code/openapi-generator/samples/server/petstore/kotlin/vertx/src/main/kotlin/org/openapitools/server/api/verticle/PetApiVertxProxyHandler.kt: (107, 40) Unresolved reference: TypeToken
[ERROR] /Users/williamcheng/Code/openapi-generator/samples/server/petstore/kotlin/vertx/src/main/kotlin/org/openapitools/server/api/verticle/PetApiVertxProxyHandler.kt: (107, 57) Unresolved reference: List
[ERROR] /Users/williamcheng/Code/openapi-generator/samples/server/petstore/kotlin/vertx/src/main/kotlin/org/openapitools/server/api/verticle/PetApiVertxProxyHandler.kt: (107, 82) Unresolved reference: type
[ERROR] /Users/williamcheng/Code/openapi-generator/samples/server/petstore/kotlin/vertx/src/main/kotlin/org/openapitools/server/api/verticle/PetApiVertxProxyHandler.kt: (110, 39) Unresolved reference: JsonArray
[ERROR] /Users/williamcheng/Code/openapi-generator/samples/server/petstore/kotlin/vertx/src/main/kotlin/org/openapitools/server/api/verticle/PetApiVertxProxyHandler.kt: (121, 24) Unresolved reference: tags
[ERROR] /Users/williamcheng/Code/openapi-generator/samples/server/petstore/kotlin/vertx/src/main/kotlin/org/openapitools/server/api/verticle/PetApiVertxProxyHandler.kt: (125, 40) Unresolved reference: TypeToken
[ERROR] /Users/williamcheng/Code/openapi-generator/samples/server/petstore/kotlin/vertx/src/main/kotlin/org/openapitools/server/api/verticle/PetApiVertxProxyHandler.kt: (125, 57) Unresolved reference: List
[ERROR] /Users/williamcheng/Code/openapi-generator/samples/server/petstore/kotlin/vertx/src/main/kotlin/org/openapitools/server/api/verticle/PetApiVertxProxyHandler.kt: (125, 82) Unresolved reference: type
[ERROR] /Users/williamcheng/Code/openapi-generator/samples/server/petstore/kotlin/vertx/src/main/kotlin/org/openapitools/server/api/verticle/PetApiVertxProxyHandler.kt: (128, 39) Unresolved reference: JsonArray
[ERROR] /Users/williamcheng/Code/openapi-generator/samples/server/petstore/kotlin/vertx/src/main/kotlin/org/openapitools/server/api/verticle/PetApiVertxProxyHandler.kt: (195, 82) Unresolved reference: file
[ERROR] /Users/williamcheng/Code/openapi-generator/samples/server/petstore/kotlin/vertx/src/main/kotlin/org/openapitools/server/api/verticle/UserApiVertxProxyHandler.kt: (85, 24) Unresolved reference: body
[ERROR] /Users/williamcheng/Code/openapi-generator/samples/server/petstore/kotlin/vertx/src/main/kotlin/org/openapitools/server/api/verticle/UserApiVertxProxyHandler.kt: (89, 40) Unresolved reference: TypeToken
[ERROR] /Users/williamcheng/Code/openapi-generator/samples/server/petstore/kotlin/vertx/src/main/kotlin/org/openapitools/server/api/verticle/UserApiVertxProxyHandler.kt: (89, 57) Unresolved reference: List
```
Does it work for you locally in your machine?

**Comment by Wooyme** [comment]
fixed it

File: modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/KotlinVertxServerCodegen.java [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+    public static final String PROJECT_NAME = "projectName";
+
+    static Logger LOGGER = LoggerFactory.getLogger(KotlinVertxServerCodegen.class);
+
+    public CodegenType getTag() {
+        return CodegenType.SERVER;
+    }
+
+    public String getName() {
+        return "kotlin-vertx";
```
Comment (wing328) [inline modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/KotlinVertxServerCodegen.java]: Heads up: I'll rename it as "kotlin-vertx-server" to follow the naming convention.

**Comment by wing328** [comment]
Found some warnings:
```
[WARNING] Source root doesn't exist: /Users/williamcheng/Code/openapi-generator/samples/server/petstore/kotlin/vertx/src/main/java
[WARNING] Using experimental Kotlin incremental compilation
[WARNING] /Users/williamcheng/Code/openapi-generator/samples/server/petstore/kotlin/vertx/src/main/kotlin/org/openapitools/server/api/verticle/PetApi.kt: (15, 8) This class shouldn't be used in Kotlin. Use kotlin.collections.List or kotlin.collections.MutableList instead.
[WARNING] /Users/williamcheng/Code/openapi-generator/samples/server/petstore/kotlin/vertx/src/main/kotlin/org/openapitools/server/api/verticle/PetApi.kt: (16, 8) This class shouldn't be used in Kotlin. Use kotlin.collections.Map or kotlin.collections.MutableMap instead.
[WARNING] /Users/williamcheng/Code/openapi-generator/samples/server/petstore/kotlin/vertx/src/main/kotlin/org/openapitools/server/api/verticle/PetApi.kt: (56, 53) Unchecked cast: Any! to Map<String, Any>
[WARNING] /Users/williamcheng/Code/openapi-generator/samples/server/petstore/kotlin/vertx/src/main/kotlin/org/openapitools/server/api/verticle/PetApi.kt: (56, 56) This class shouldn't be used in Kotlin. Use kotlin.collections.Map or kotlin.collections.MutableMap instead.
[WARNING] /Users/williamcheng/Code/openapi-generator/samples/server/petstore/kotlin/vertx/src/main/kotlin/org/openapitools/server/api/verticle/PetApiVertxProxyHandler.kt: (42, 31) Parameter 'id' is never used
[WARNING] /Users/williamcheng/Code/openapi-generator/samples/server/petstore/kotlin/vertx/src/main/kotlin/org/openapitools/server/api/verticle/StoreApi.kt: (14, 8) This class shouldn't be used in Kotlin. Use kotlin.collections.List or kotlin.collections.MutableList instead.
[WARNING] /Users/williamcheng/Code/openapi-generator/samples/server/petstore/kotlin/vertx/src/main/kotlin/org/openapitools/server/api/verticle/StoreApi.kt: (15, 8) This class shouldn't be used in Kotlin. Use kotlin.collections.Map or kotlin.collections.MutableMap instead.
[WARNING] /Users/williamcheng/Code/openapi-generator/samples/server/petstore/kotlin/vertx/src/main/kotlin/org/openapitools/server/api/verticle/StoreApi.kt: (43, 53) Unchecked cast: Any! to Map<String, Any>
[WARNING] /Users/williamcheng/Code/openapi-generator/samples/server/petstore/kotlin/vertx/src/main/kotlin/org/openapitools/server/api/verticle/StoreApi.kt: (43, 56) This class shouldn't be used in Kotlin. Use kotlin.collections.Map or kotlin.collections.MutableMap instead.
[WARNING] /Users/williamcheng/Code/openapi-generator/samples/server/petstore/kotlin/vertx/src/main/kotlin/org/openapitools/server/api/verticle/StoreApiVertxProxyHandler.kt: (41, 31) Parameter 'id' is never used
[WARNING] /Users/williamcheng/Code/openapi-generator/samples/server/petstore/kotlin/vertx/src/main/kotlin/org/openapitools/server/api/verticle/UserApi.kt: (14, 8) This class shouldn't be used in Kotlin. Use kotlin.collections.List or kotlin.collections.MutableList instead.
[WARNING] /Users/williamcheng/Code/openapi-generator/samples/server/petstore/kotlin/vertx/src/main/kotlin/org/openapitools/server/api/verticle/UserApi.kt: (15, 8) This class shouldn't be used in Kotlin. Use kotlin.collections.Map or kotlin.collections.MutableMap instead.
[WARNING] /Users/williamcheng/Code/openapi-generator/samples/server/petstore/kotlin/vertx/src/main/kotlin/org/openapitools/server/api/verticle/UserApi.kt: (55, 53) Unchecked cast: Any! to Map<String, Any>
[WARNING] /Users/williamcheng/Code/openapi-generator/samples/server/petstore/kotlin/vertx/src/main/kotlin/org/openapitools/server/api/verticle/UserApi.kt: (55, 56) This class shouldn't be used in Kotlin. Use kotlin.collections.Map or kotlin.collections.MutableMap instead.
[WARNING] /Users/williamcheng/Code/openapi-generator/samples/server/petstore/kotlin/vertx/src/main/kotlin/org/openapitools/server/api/verticle/UserApiVertxProxyHandler.kt: (41, 31) Parameter 'id' is never used
```

File: modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/KotlinVertxServerCodegen.java (hunk 4 above)
Comment (wing328) [reply modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/KotlinVertxServerCodegen.java]: Second though: kotlin-vertx is fine given that we've kotlin-spring

**Comment by wing328** [comment]
@Wooyme thanks for the PR, which has been included in the v4.1.3 release: https://twitter.com/oas_generator/status/1180123829626003456

**Pull request merged**